autoload/airline/extensions/ale.vim

```diff
@@ -8,6 +8,27 @@ function! s:airline_ale_count(cnt, symbol)
   return a:cnt ? a:symbol. a:cnt : ''
 endfunction
 
+function! s:airline_ale_get_line_number(cnt, type) abort
+  if a:cnt == 0
+    return ''
+  endif
+
+  let problem_type = (a:type ==# 'error') ? 'E' : 'W'
+  let buffer       = bufnr('')
+  let problems     = copy(ale#engine#GetLoclist(buffer))
+  call filter(problems, 'v:val.type =~ "' . problem_type . '"')
```

Comment:
You need to move the buffer number check into the `filter` here. Item 0 could be a problem for some other buffer, and there could be an item for the current buffer later on in the List.